test/known_issues/test-assert-deepEqual-circular.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+require('../common');
+
+// Refs: https://github.com/nodejs/node/issues/14441
+
+const assert = require('assert');
+
+{
+  const a = {};
+  const b = {};
+  a.a = a;
+  b.a = {};
+  b.a.a = a;
+  assert.deepEqual(b, a); // eslint-disable-line no-restricted-properties
+  assert.deepEqual(a, b); // eslint-disable-line no-restricted-properties
+}
+
+{
+  const a = new Set();
+  const b = new Set();
+  const c = new Set();
+  a.add(a);
+  b.add(b);
+  c.add(a);
+  assert.deepEqual(c, b); // eslint-disable-line no-restricted-properties
+  assert.deepEqual(b, c); // eslint-disable-line no-restricted-properties
```

Comment:
It is a set that has a single entry: a set that has a set as only entry that is circular. And that should be the same as a set that has itself as single entry.

I already have the fix for the issue but I am still in the office and I will open a PR with the fix for these later on.